csv.h

```diff
@@ -462,9 +491,14 @@ namespace io{
                                 }
                         }
 
+						bool quoted_data = false;
                         int line_end = data_begin;
-                        while(buffer[line_end] != '\n' && line_end != data_end){
+                        while( (quoted_data || buffer[line_end] != '\n') && line_end != data_end){
+							if (hasQuote && buffer[line_end] == quoteChar) {
+								quoted_data = !quoted_data;
```

Comment:
How does this work, if somebody has a backslash-quote-policy? As far as I see, you are counting the number of quotes. This happens to work for double quote escape but fails for backslash quote escape.